index.py

```diff
@@ -502,52 +502,52 @@ def update_suggestions(n_intervals, n_submit, n_clicks, value):
     return suggestions, no_update, True
 
 # Completion clicked
-# TODO: convert to clientside callback, if pattern matching is supported for them
-@app.callback(
-    Output('search_bar_input', 'value'),
+clientside_callback(
+    """
+    function on_completion(n_clicks) {
+        const ctx = dash_clientside.callback_context;
+        let triggered_id = ctx.triggered[0].prop_id;
+
+        if (!ctx.triggered[0].value)
+            return dash_clientside.no_update;
+
+        if (triggered_id.search('.n_clicks') > 0) {
+            triggered_id = JSON.parse(triggered_id.substr(0, triggered_id.indexOf('.n_clicks')));
+            return triggered_id.text + ' ';
+        }
+        return dash_clientside.no_update;
+    }
+    """,
+    Output('search_bar_input', 'value', allow_duplicate=True),
     Input({'type': 'search_bar_completion', 'index': ALL, 'text': ALL}, 'n_clicks'),
     prevent_initial_call=True
 )
-def on_completion(n_clicks):
-    ctx = dash.callback_context
-
-    if ctx.triggered[0]['value']:
-        return ctx.triggered_id['text'] + ' '
-
-    return no_update
 
 # Quick field clicked
-# clientside_callback(
-#     """
-#     function on_quickfield(n_clicks, values, value) {
-#         const ctx = dash_clientside.callback_context;
-#         window['ctx'] = ctx;
-#         return window.dash_clientside.callback_context.triggered[0].prop_id.split('.')[0];
-#     }
-#     """,
-#     Output('search_bar_input', 'value'),
-#     Input({'type': 'search_bar_quick_field', 'index': ALL}, 'n_clicks'),
-#     State({'type': 'search_bar_quick_field', 'index': ALL}, 'children'),
-#     State('search_bar_input', 'value'),
-#     prevent_initial_call=True
-# )
-
-# TODO: convert to clientside callback, if pattern matching is supported for them
-@app.callback(
+clientside_callback(
```

Comment:
The quickfields do not work anymore. They are clickable, but the searchbar is not updated.